tests/functional/afr/test_self_heal_with_full_algorithm.py

```diff
@@ -0,0 +1,207 @@
+        self.get_super_method(self, 'tearDown')()
+
+    def test_self_heal_with_full_algorithm(self):
+        """
+        Test Steps:
+        1. Create a replicated/distributed-replicate volume and mount it
+        2. Set data/metadata/entry-self-heal to off and
+           data-self-heal-algorithm to full
+        3. Create a directory from the mount point
+        4. Bring down all bricks from a selected set
+        5. Create few files inside the directory with some data
+        6. Bring back the bricks online and wait for heal to complete
+        7. Check arequal of the subvol and all the bricks in the subvol should
+           have same checksum
+        8. Bring down all bricks from another set
+        9. Modify the data of existing files under the directory
+        10. Bring back the bricks online and wait for heal to complete
+        11. Check arequal of the subvol and all the brick in the same subvol
+            should have same checksum
+        """
+
+        # Setting options
+        for key, value in (("data-self-heal", "off"),
+                           ("metadata-self-heal", "off"),
+                           ("entry-self-heal", "off"),
+                           ("data-self-heal-algorithm", "full")):
+            ret = set_volume_options(self.mnode, self.volname, {key: value})
+            self.assertTrue(ret, 'Failed to set %s to %s.' % (key, value))
+            g.log.info("%s set to %s successfully", key, value)
+
+        # Create a directory under the mount point
+        mountpoint = self.mounts[0].mountpoint
+        client = self.mounts[0].client_system
+
+        cmd = ("mkdir %s/test_full_self_heal" % (mountpoint))
+        ret, _, _ = g.run(client, cmd)
+        self.assertEqual(ret, 0, "Failed to create directory on mountpoint")
+        g.log.info("Successfully created directory on mountpoint")
+
+        # List a brick in each subvol and bring them offline
+        brick_to_bring_offline = []
+        subvols = get_subvols(self.mnode, self.volname)['volume_subvols']
+        for subvol in subvols:
+            self.assertTrue(subvol, "List is empty")
+            brick_to_bring_offline.extend(sample(subvol, 1))
+
+        ret = bring_bricks_offline(self.volname, brick_to_bring_offline)
+        self.assertTrue(ret, "Unable to bring brick: {} offline".format(
+            brick_to_bring_offline))
+
+        # Validate the bricks are offline
+        ret = are_bricks_offline(self.mnode, self.volname,
+                                 brick_to_bring_offline)
+        self.assertTrue(ret, "Brick:{} is still online".format(
+            brick_to_bring_offline))
+
+        # Create few files under the directory with data
+        cmd = ("cd %s/test_full_self_heal ; for i in `seq 1 100` ; "
+               "do dd if=/dev/urandom of=file.$i bs=1M count=1; done;"
+               % (mountpoint))
+        ret, _, _ = g.run(client, cmd)
+        self.assertEqual(ret, 0, "Failed to create file on mountpoint")
```

Comment:
In general, there are many observations if cmds are chained (or not) `ret` is `0` even if cmd is failed and better to `assertFalse` on stderr.

Reply:
Done.